lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Addresses.hs

```diff
@@ -253,6 +257,83 @@ spec = describe "SHELLEY_ADDRESSES" $ do
                 (`shouldNotSatisfy` T.isPrefixOf "addr_test")
             ]
 
+    it "ADDRESS_LIST_06 - Change addresses are listed even after transaction is no longer pending" $ \ctx -> runResourceT $ do
+        let initPoolGap = 10
+
+        let verifyAddrs i p addr = do
+                liftIO (length (filter ((== Used) . (^. (#state . #getApiT))) addr)
+                    `shouldBe` p)
+                liftIO (length addr `shouldBe` i)
+
+        -- 1. create shelley wallet
+        wSrc <- fixtureWallet ctx
+        wDest <- emptyWalletWith ctx ("Wallet", "cardano-wallet", initPoolGap)
+
+        -- 2. list addresses of that wallet (for later comparison)
+        listAddresses @n ctx wSrc >>= verifyAddrs 50 10
+
+        -- 3. send a transaction from that wallet
+        addrs <- listAddresses @n ctx wDest
+        verifyAddrs 20 0 addrs
```

Comment:
that's what I expect given `initPoolGap` ie., 10 Unused change addresses and 10 Unused external addresses